File: backend/src/main/java/project1/service/SkillService.java
package project1.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import project1.entity.Skills;
import project1.repository.SkillRepo;

@Service
public class SkillService {

    @Autowired
    SkillRepo repository;

    public List<Skills> getAllSkills() {
        return repository.findAll();
    }

    public Skills getSkillById(long id) {
        return repository.findById(id).orElse(null);
    }

    public Skills createSkill(Skills skill) {
        return repository.save(skill);
    }

    public Skills updateSkill(Skills skill) {
        return repository.save(skill);
    }

    public String removeSkill(long id) {
        repository.deleteById(id);
        return "Skill Deleted Successfully";
    }
}

File: frontend/src/services/skillService.js
import api from "./api";

export const skillService = {

    // Get All Skills
    async getAllSkills() {

        const response = await api.get("/getallskills");

        return response.data;

    },

    // Get Skill By ID
    async getSkillById(id) {

        const response = await api.get(`/skill/${id}`);

        return response.data;

    },

    // Create Skill
    async createSkill(skill) {

        const response = await api.post("/skill/create", skill);

        return response.data;

    },

    // Update Skill
    async updateSkill(skill) {

        const response = await api.put("/skill/update", skill);

        return response.data;

    },

    // Delete Skill
    async deleteSkill(id) {

        const response = await api.delete(`/skill/delete/${id}`);

        return response.data;

    }

};